app/models/submission.rb

```diff
@@ -482,6 +498,268 @@ def turnitinable_by_lti?
     turnitin_data.select{|_, v| v.is_a?(Hash) && v.key?(:outcome_response)}.any?
   end
 
+  # VeriCite
+
+  # this function will check if the score needs to be updated and update/save the new score if so,
+  # otherwise, it just returns the vericite_data_hash
+  def vericite_data()
+    self.vericite_data_hash ||= {}
+    self.vericite_data_hash[:provider] = :vericite
+    # check to see if the score is stale, if so, fetch it again
+    updateScores = false
+    # since there could be multiple versions, let's not waste calls for old versions and use the old score
+    if !self.readonly? && self.versions.current.number == self.version_number
+      self.vericite_data_hash.keys.each do |asset_string|
+        data = self.vericite_data_hash[asset_string]
+        next unless data && data.is_a?(Hash) && data[:object_id]
+        updateScores = updateScores || vericite_recheck_score(data)
+      end
+      # we have found at least one score that is stale, call VeriCite and save the results
+      if updateScores
+        check_vericite_status
+      end
```

Comment:
This call requests every single score for the entire assignment in one call and then cache's the results for the subsequent requests. This can only result in one network call for a page load. There is an additional safety measure that makes sure the score's cache is only called once every 5 minutes at a maximum for any assignment no matter which user requests it.